fedimint/src/bin/configgen.rs

```diff
@@ -5,22 +5,41 @@ use fedimint_api::{Amount, PeerId};
 use rand::rngs::OsRng;
 use std::path::PathBuf;
 
+/// Config generator for Fedimint Federation
+///
+/// Running this program will generate config
+/// files for federation member nodes in directory
+/// specified with `out-dir`
 #[derive(Parser)]
 struct Options {
-    cfg_path: PathBuf,
-    nodes: u16,
+    /// Directory to output all the generated config files
+    #[clap(long = "out-dir")]
+    dir_out_path: PathBuf,
+
+    /// Number of nodes in the federation
+    #[clap(long = "num-nodes")]
+    num_nodes: u16,
+
+    /// Base hbbft port
+    #[clap(long = "hbbft-base-port")]
     hbbft_base_port: u16,
+
+    /// Base api port
+    #[clap(long = "api-base-port")]
     api_base_port: u16,
-    amount_tiers: Vec<Amount>,
+
+    /// Available denominations of notes issues by the federation (comma separated)
+    #[clap(long = "denominations", value_delimiter = ',')]
```

Comment:
Can we enforce that there has to be at least one denomination? Forgetting to specify them bit me a few times …

I would have assumed that this is the default behavior, but apparently it isn't (I just tested it again) and there has been a lengthy discussion about it (didn't read it all) https://github.com/clap-rs/clap/issues/1772